CRM/Contribute/Form/UpdateBilling.php

```diff
@@ -47,14 +47,14 @@ public function preProcess() {
     if ($this->_mid) {
       $this->_paymentProcessor = CRM_Financial_BAO_PaymentProcessor::getProcessorForEntity($this->_mid, 'membership', 'info');
       $this->_paymentProcessor['object'] = CRM_Financial_BAO_PaymentProcessor::getProcessorForEntity($this->_mid, 'membership', 'obj');
-      $this->_subscriptionDetails = CRM_Contribute_BAO_ContributionRecur::getSubscriptionDetails($this->_mid, 'membership');
+      $this->setSubscriptionDetails();
```

Comment:
I don't think this works - `setSubscriptionDetails` was already called when it called `parent::preProcess`  - so either that worked & this is unnecessary or it didn't work & this won't help

That code kinda does my head in but I think the idea should be

-either we have the contribution_recur_id or we need to load it from the contribution or membership.
- either we have the membership id or we need to load it from the contribution or recurring contribution

shit just got circular.

Reply:
@eileenmcnaughton Well spotted! I just swapped the line for the "equivalent" but have now looked in more detail and you are right. It's already called in `preProcess()` *and* we already have retrieved all the IDs by that point. Also the `getSubscriptionDetails()` function is only using the different entity IDs to work out the recur ID - so the same data is returned no matter what related entity is passed in.

So I've now just removed this extra call to get/set the subscription details.